src/Compilers/VisualBasic/Portable/Compilation/VisualBasicDeterministicKeyBuilder.vb

```diff
@@ -0,0 +1,100 @@
+                Throw New InvalidOperationException()
+            End If
+
+            MyBase.WriteParseOptionsCore(writer, parseOptions)
+
+            writer.Write("languageVersion", basicOptions.LanguageVersion)
+            writer.Write("specifiedLanguageVersion", basicOptions.SpecifiedLanguageVersion)
+
+            writer.WriteKey("preprocessorSymbols")
+            writer.WriteObjectStart()
+            For Each pair In basicOptions.PreprocessorSymbols.OrderBy(Function(x, y) StringComparer.Ordinal.Compare(x.Key, y.Key))
+                Dim value = pair.Value
+                If value Is Nothing Then
+                    writer.WriteNull(pair.Key)
+                    Continue For
+                End If
+
+                writer.WriteKey(pair.Key)
+
+                Dim type = value.GetType()
+                If type = GetType(String) Then
+                    writer.Write(CType(value, String))
+                ElseIf type = GetType(Boolean) Then
+                    writer.Write(CType(value, Boolean))
+                Else
+                    Dim formattable = TryCast(value, IFormattable)
+                    If formattable IsNot Nothing Then
+                        writer.Write(formattable.ToString(Nothing, CultureInfo.InvariantCulture))
+                    Else
+                        Throw ExceptionUtilities.UnexpectedValue(value)
+                    End If
+                End If
+            Next
+            writer.WriteObjectEnd()
+
+        End Sub
+
+        Protected Overrides Sub WriteCompilationOptionsCore(writer As JsonWriter, options As CompilationOptions)
+            Dim basicOptions = TryCast(options, VisualBasicCompilationOptions)
+            If basicOptions Is Nothing Then
+                Throw New InvalidOperationException()
+            End If
+
+            MyBase.WriteCompilationOptionsCore(writer, options)
+
+            writer.Write("rootNamespace", basicOptions.RootNamespace)
+            writer.Write("optionStrict", basicOptions.OptionStrict)
+            writer.Write("optionInfer", basicOptions.OptionInfer)
+            writer.Write("optionExplicit", basicOptions.OptionExplicit)
+            writer.Write("optionCompareText", basicOptions.OptionCompareText)
+            writer.Write("embedVbCoreRuntime", basicOptions.EmbedVbCoreRuntime)
+
+            writer.WriteKey("globalImports")
+            writer.WriteArrayStart()
+            For Each import In basicOptions.GlobalImports
+                writer.WriteObjectStart()
+                writer.Write("name", import.Name)
+                writer.Write("isXmlClause", import.IsXmlClause)
+                writer.WriteObjectEnd()
+            Next
+            writer.WriteArrayEnd()
+
+            writer.WriteKey("parseOptions")
+            If basicOptions.ParseOptions IsNot Nothing Then
+                WriteParseOptions(writer, basicOptions.ParseOptions)
+            Else
+                writer.WriteObjectStart()
```

Comment:
Should this be `WriteNull` instead?

Reply:
This is the last open question

Reply:
I saw you marked this as resolved, but I'm not sure what the resolution was. Could you clarify?

Reply:
I swear I flipped that to `WriteNull` as I marked it resolved but it's not showing up so clearly I didn't. Let me try a second time 😦

Reply:
Oh I did add it but i did not push 🤦